pkg/app/ops/deploymentchaincontroller/updater.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploymentchaincontroller
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+// updater watches for a specified deployment model object
+// and updates the state of that object based on states of
+// its deployments.
+type updater struct {
+	deploymentChainID string
+	// applicationsRef contains list of all applications of
+	// the current handling deployment chain.
+	applicationsRef []*model.ChainApplicationRef
+	// deploymentsRef is a map with key is the application id and
+	// value is the last state of deployment ref to the deployment
+	// of that in chain application.
+	deploymentsRef map[string]*model.ChainDeploymentRef
```

Comment:
I feel like they should be respectively:
-> applicationRefs
-> deploymentRefs

The word `reference` can be plural so possibly it's more clear

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2986/commits/1868da8888d7049bb4804b98f5f9481ba494235a